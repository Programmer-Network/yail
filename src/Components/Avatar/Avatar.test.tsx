import { render, screen } from "@testing-library/react";

import { Avatar } from "./";

describe("Avatar component", () => {
  test("renders an image when src is provided", () => {
    render(
      <Avatar
        alt='Test Avatar'
        src='https://avatars.githubusercontent.com/u/2085263?v=4'
      />
    );
    const image = screen.getByRole("img", { name: "Test Avatar" });
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute(
      "src",
      "https://avatars.githubusercontent.com/u/2085263?v=4"
    );
  });

  test("renders a BoringAvatars SVG when src is not provided", () => {
    render(<Avatar />);
    const avatarParent = screen.getByTestId("svg-avatar-parent");
    const svg = avatarParent.querySelector("svg");
    expect(svg).toBeInTheDocument();
  });

  test("applies custom size and colors", () => {
    const customColors = ["#FF5733", "#33FF57"];
    render(<Avatar size={100} colors={customColors} />);
  });

  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(<Avatar alt='Test Avatar' />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("throws an error for invalid URL", () => {
    const invalidSrc = "invalid-url";
    expect(() => render(<Avatar src={invalidSrc} />)).toThrow(
      "Invalid URL provided"
    );
  });
});
