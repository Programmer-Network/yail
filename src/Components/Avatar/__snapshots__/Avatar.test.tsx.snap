// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <div
    class="yl:sm:mt-0 yl:rounded-full yl:overflow-hidden yl:flex yl:items-center yl:justify-center yl:flex-shrink-0"
    data-testid="svg-avatar-parent"
    style="width: 50px; height: 50px;"
  >
    <svg
      fill="none"
      height="50"
      role="img"
      viewBox="0 0 36 36"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <mask
        height="36"
        id="«r2»"
        maskUnits="userSpaceOnUse"
        width="36"
        x="0"
        y="0"
      >
        <rect
          fill="#FFFFFF"
          height="36"
          rx="72"
          width="36"
        />
      </mask>
      <g
        mask="url(#«r2»)"
      >
        <rect
          fill="#ad46ff"
          height="36"
          width="36"
        />
        <rect
          fill="#ffb900"
          height="36"
          rx="6"
          transform="translate(7 1) rotate(73 18 18) scale(1.1)"
          width="36"
          x="0"
          y="0"
        />
        <g
          transform="translate(3.5 -6) rotate(-3 18 18)"
        >
          <path
            d="M15 20c2 1 4 1 6 0"
            fill="none"
            stroke="#000000"
            stroke-linecap="round"
          />
          <rect
            fill="#000000"
            height="2"
            rx="1"
            stroke="none"
            width="1.5"
            x="11"
            y="14"
          />
          <rect
            fill="#000000"
            height="2"
            rx="1"
            stroke="none"
            width="1.5"
            x="23"
            y="14"
          />
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
